core/src/models/Destination.ts

```diff
@@ -240,6 +300,46 @@ export class Destination extends LoggedModel<Destination> {
     return DestinationOps.unTrackGroup(this);
   }
 
+  async getSupportedSyncModes() {
+    return DestinationOps.getSupportedSyncModes(this);
+  }
+
+  async getSyncMode() {
+    if (this.syncMode) return this.syncMode;
+
+    const supportedSyncModes = await this.getSupportedSyncModes();
+    if (supportedSyncModes.length === 1) {
+      // If destination only supports one sync mode, use it
+      return supportedSyncModes[0];
+    } else if (supportedSyncModes.length === 0) {
+      // If destination does not support sync modes, let it create, update, and delete
+      return "sync";
+    }
+
+    return null;
+  }
```

Comment:
You can overload a Sequelize Model's getter and setting, which I think this is what you are trying to do.  In this way, you can then reference `destination.syncMode` directly.  

Check out how we modify `Import.data` with a custom getter and setter here to do some JSON parsing.  https://github.com/grouparoo/grouparoo/blob/main/core/src/models/Import.ts#L58-L67

Reply:
Turns out you can't - async